src/jdkInstaller.ts

```diff
@@ -0,0 +1,97 @@
+import * as path from "path";
+import * as request from "request";
+import * as fs from "fs";
+import * as mkdirp from "mkdirp";
+import * as pcp from "promisify-child-process";
+import { OutputChannel } from "vscode";
+
+export class JdkInstaller {
```

Comment:
I would prefer to refactor this not to be a class.

The only purpose of this class is to close over `out` which can simply be a parameter of `installJava` and then we can have a single function. All other class members can be simple values/functions inside this module.

I can work on this refactor @tgodizk